tests/epyccel/test_imports.py

```diff
@@ -0,0 +1,60 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+
+import pytest
+from numpy import ones
+
+from pyccel.epyccel import epyccel
+from pyccel.decorators import types
+
+#------------------------------------------------------------------------------
+@pytest.fixture(params=[
+    pytest.param('python', marks = pytest.mark.python)]
+)
+def language(request):
+    return request.param
+
+#==============================================================================
+
+def test_import(language):
+    @types('int[:]')
+    def f1(x):
+        import numpy
+        s = numpy.shape(x)[0]
+        return s
+
+    f = epyccel(f1, language = language)
+    x = ones(10, dtype=int)
+    assert f(x) == f1(x)
+
+def test_import_from(language):
+    @types('int[:]')
+    def f2(x):
+        from numpy import shape
+        s = shape(x)[0]
+        return s
+
+
+    f = epyccel(f2, language = language)
+    x = ones(10, dtype=int)
+    assert f(x) == f2(x)
+
+def test_import_as(language):
+    @types('int[:]')
+    def f3(x):
+        import numpy as np
+        s = np.shape(x)[0]
+        return s
+
+    f = epyccel(f3, language = language)
+    x = ones(10, dtype=int)
+    assert f(x) == f3(x)
+
+def test_import_collision(language):
+    @types('int')
+    def f4(x):
+        import modules.Module_3 as mod
+        add_one = mod.add_one(x)
+        return add_one

```

Comment:
This test has "collision" in the name but it does not appear to test collisions. Or have I missed something?

Reply:
as said before the dotted element rhs translated to just a function call  add_one = add_one(x)
but in the semantic stage, we clone the function with a different name to avoid name collision.

Reply:
Ah yes sorry the function and the variable have the same name :+1: